example/Setup.hs

```diff
@@ -1,6 +1,31 @@
-module Main where
+{-# LANGUAGE CPP #-}
+{-# OPTIONS_GHC -Wall #-}
+module Main (main) where
 
-import Distribution.Extra.Doctest (defaultMainWithDoctests)
+#ifndef MIN_VERSION_cabal_doctest
+#define MIN_VERSION_cabal_doctest(x,y,z) 0
+#endif
 
+#if MIN_VERSION_cabal_doctest(1,0,0)
+
+import Distribution.Extra.Doctest ( defaultMainWithDoctests )
 main :: IO ()
 main = defaultMainWithDoctests "doctests"
+
+#else
+
+#ifdef MIN_VERSION_Cabal
+-- If the macro is defined, we have new cabal-install,
```

Comment:
I'd like `#warning` more, because it's easy to forgot to include `Warning.hs` (CI doesn't catch it), as I did with `bytes` and `distributive`.